### PR TITLE
Final Incremental Neural Operator changes

### PR DESCRIPTION
**Pull request by Robertboy18**
I'm excited to share with you the final update for the Incremental version of Neural Operators. This pull request includes the latest changes and improvements to the Incremental module, bringing it to its completed state.

Here's a brief overview of the updates included in this pull request:

1. Algorithm 1: Incremental
  
Added a new method called loss_gap(loss) to handle the incremental mode increase based on the loss gap. The method tracks the loss values and increases the incremental modes if the difference between consecutive losses falls below a predefined threshold.

2. Algorithm 2: Gradient-based Explained Ratio

Implemented the grad_explained() method to handle the incremental mode increase based on the explained variance of gradients. The method accumulates gradients and checks the explained variance of each mode. The incremental mode is increased if the explained variance is below a specified threshold.

3. Algorithm 3: Regularize Input Resolution

Added the incremental_resolution_regularize(x, y) method to regulate the input resolution based on the dataset being used.
The method applies sub-sampling to the input data to reduce the resolution, depending on the current index and dataset.

4. Other improvements

Added extra data loaders for some datasets, incremental config files as well as multiple tests for the algorithm and a test example for the users to understand. Added necessary checks to ensure that the incremental_grad and incremental_loss_gap flags cannot be used together. Updated comments throughout the code to provide clarity and improve readability. 

Looking forward to your feedback and suggestions.

### CODE REVIEW

**Review by JeanKossaifi — COMMENTED** [review]
Overall, it's good. Let's try to make it a little leaner and decouple the algorithm from the data. I don't think we need to have the Parameter class for this.

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +0,0 @@
-wandb 
-ruamel.yaml
-configmypy
-tensorly
-tensorly-torch
-opt-einsum
-h5py
-zarr
```
[the remark this answers is not in the document]
Comment (Robertboy18) [reply requirements.txt:8]: Yes, for sure! Sorry for removing it.

File: neuralop/training/paramaters.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            self.loss_list = []
+        
+        if self.incremental_resolution:
+            # incremental resolution
+            paramaters_resolution = paramaters.incremental_resolution
+            self.epoch_gap = paramaters_resolution.epoch_gap
+            
+            # Determine the sub_list based on the dataset_name
+            # Best to do powers of two as FFT is faster and FFTW++ library optimizes for powers of two
+            if self.dataset_name == 'SmallDarcy':
```
[the remark this answers is not in the document]
Comment (Robertboy18) [reply neuralop/training/paramaters.py:40]: Yes, for sure understandable! I will push the new changes soon!

File: scripts/train_navier_stokes.py [hunk 3]
```diff
@@ -32,9 +32,9 @@
     else:
         wandb_name = '_'.join(
             f'{var}' for var in [config_name, config.tfno2d.n_layers, config.tfno2d.modes_width, config.tfno2d.modes_height,
-                                 config.tfno2d.width, config.tfno2d.factorization, config.tfno2d.rank, 
+                                 config.tfno2d.hidden_channels, config.tfno2d.factorization, config.tfno2d.rank, 
                                  config.patching.levels, config.patching.padding])
-    wandb.init(config=config, name=wandb_name, group=config.wandb.group,
```
[the remark this answers is not in the document]
Comment (Robertboy18) [reply scripts/train_navier_stokes.py:37]: Oh sorry I didnt have a group config so removed it as I got an error, will add it back!

File: scripts/train_navier_stokes.py [hunk 4]
```diff
@@ -131,7 +131,7 @@
                   log_test_interval=config.wandb.log_test_interval,
                   log_output=config.wandb.log_output,
                   use_distributed=config.distributed.use_distributed,
-                  verbose=config.verbose)
+                  verbose=config.verbose, incremental = True)
```
[the remark this answers is not in the document]
Comment (Robertboy18) [reply scripts/train_navier_stokes.py:134]: Sure, will do an auto pep8 update to all files that I updated!

**Pull request closed without merging**